internals/plan/plan.go

```diff
@@ -535,6 +535,7 @@ func (e *FormatError) Error() string {
 // CombineLayers combines the given layers into a single layer, with the later
 // layers overriding earlier ones.
 func CombineLayers(layers ...*Layer) (*Layer, error) {
+	var errorMessages []string // To collect multiple errors
```

Comment:
I wonder if we can avoid the `errorMessages = append(errorMessages, ...)` boilerplate by doing the following:

```go
var errorMessages []string
addError := func(message string) {
    errorMessages = append(errorMessages, message)
}
```

Or even:

```go
var errorMessages []string
addErrorf := func(format string, args ...any) {
    errorMessages = append(errorMessages, fmt.Sprintf(format, args...))
}
```